ouroboros-consensus/src/Ouroboros/Consensus/MiniProtocol/ChainSync/Server.hs

```diff
@@ -44,42 +41,6 @@ chainSyncBlockServerFollower
     -> m (Follower m blk (WithPoint blk (Serialised blk)))
 chainSyncBlockServerFollower chainDB registry = ChainDB.newFollower chainDB registry getSerialisedBlockWithPoint
 
--- | Chain Sync Server for block headers for a given a 'ChainDB'.
---
--- The node-to-node protocol uses the chain sync mini-protocol with chain
--- headers (and fetches blocks separately with the block fetch mini-protocol).
---
-chainSyncHeadersServer
-    :: forall m blk.
-       ( IOLike m
-       , HasHeader (Header blk)
-       )
-    => Tracer m (TraceChainSyncServerEvent blk)
-    -> ChainDB m blk
-    -> Follower m blk (WithPoint blk (SerialisedHeader blk))
-    -> NodeToNodeVersion
-    -> ChainSyncServer (SerialisedHeader blk) (Point blk) (Tip blk) m ()
-chainSyncHeadersServer tracer chainDB flr _version =
-    ChainSyncServer $
-      let ChainSyncServer server = chainSyncServerForFollower tracer chainDB flr in
-      server
-
--- | Chain Sync Server for blocks for a given a 'ChainDB'.
---
--- The local node-to-client protocol uses the chain sync mini-protocol with
```

Comment:
Would you move this comment sentence to `chainSyncServerForFollower`? It'd be good to put the two driving examples there motivating why `b` does not equal `blk`.